lineplot/lineplot.app.js

```diff
@@ -0,0 +1,231 @@
+
+                        var plots = {
+                          data: []
+                        };
+                        var tracesComplete = 0;
+                        lineplotConfig.traces.forEach(function (trace) {
+                            var uriWithFilters = baseUri + trace.path + "/subject_id=" + $rootScope.subject_id + "&recorded_time::geq::" + UriUtils.fixedEncodeURIComponent(timestamp);
+                            if ($rootScope.duration) {
+                                var end_x = moment(timestamp).add($rootScope.duration, 'h').format(dataFormats.datetime.submission);
+                                uriWithFilters += "&recorded_time::leq::" + UriUtils.fixedEncodeURIComponent(end_x);
+                            }
+                            var uri = uriWithFilters + "/" + trace.x_col + "," + trace.y_col + "@sort(recorded_time)?limit=" + $rootScope.limit;
+                            $http.get(uri).then(function(response) {
+                                // console.log(response, response.headers('content-type'));
+                                var layout = {
+                                    title: lineplotConfig.plot_title,
+                                    xaxis: {
+                                        title: lineplotConfig.x_axis_label
+                                    },
+                                    yaxis: {
+                                        title: lineplotConfig.y_axis_label
+                                    }
+                                }
+
+                                var config = {
+                                  displaylogo: false,
+                                }
+
+                                var lineplot = {
+                                    x: [],
+                                    y: [],
+                                    type: "scatter",
+                                    name: trace.label,
+                                    mode: 'lines'
+                                }
+
+                                var data = response.data;
+                                // data points taken 1 per second, 60 per minute, 3600 per hour
+                                data.forEach(function (row) {
+                                    lineplot.x.push(row[trace.x_col]);
+                                    lineplot.y.push(row[trace.y_col]);
+                                });
+
+                                plots.data.push(lineplot);
+                                plots.layout = layout;
+                                plots.config = config;
+
+
+                                tracesComplete++;
+                                if (tracesComplete == lineplotConfig.traces.length) {
+                                    $rootScope.plots = plots;
+                                    Plotly.relayout();
+                                }
+                            }).catch(function (err) {
+                                if (err.data) {
```

Comment:
You should use the error handling we have defined in chaise to handle this case. It looks like the behavior you are implementing is what we have in `chaise`.